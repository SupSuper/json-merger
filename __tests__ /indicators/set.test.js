const helpers = require("../../__helpers__");

describe("when merging two objects and a source property has a $set indicator", function () {

    test("containing an object it should set a property with $set.key as key and $set.value as value", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$set": {
                    "key": "$value",
                    "value": true
                }
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });

    test("containing an array it should set all properties defined in $set: [{key,value},{key,value}]", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$set": [
                    {
                        "key": "$value",
                        "value": true
                    },
                    {
                        "key": "$remove",
                        "value": true
                    }
                ]
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });

    test("it should process the value property in $set", function () {

        const object1 = {
            "a": {
                "aa": "one"
            }
        };

        const object2 = {
            "a": {
                "$set": {
                    "key": "$value",
                    "value": {
                        "$value": "two"
                    }
                }
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });
});
