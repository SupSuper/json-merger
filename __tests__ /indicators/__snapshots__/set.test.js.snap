// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $set indicator containing an array it should set all properties defined in $set: [{key,value},{key,value}] 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\",
		\\"$value\\": true,
		\\"$remove\\": true
	}
}"
`;

exports[`when merging two objects and a source property has a $set indicator containing an object it should set a property with $set.key as key and $set.value as value 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\",
		\\"$value\\": true
	}
}"
`;

exports[`when merging two objects and a source property has a $set indicator it should process the value property in $set 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"one\\",
		\\"$value\\": \\"two\\"
	}
}"
`;
